src/Umbraco.Core/Services/IContentTypeService.cs

```diff
@@ -24,8 +24,10 @@ public interface IContentTypeService : IService
         void SaveMediaTypeContainer(EntityContainer container, int userId = 0);
         EntityContainer GetContentTypeContainer(int containerId);
         EntityContainer GetContentTypeContainer(Guid containerId);
+        EntityContainer GetContentTypeContainer(string folderName, int level);

```

Comment:
This will need to return IEnumerable because there can be multiple containers with the same name at the same level